pkg/k8shandler/forwarding.go

```diff
@@ -59,28 +60,32 @@ func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config s
 		}
 	}
 
-	generator, err := forwarding.NewConfigGenerator(
-		clusterRequest.Cluster.Spec.Collection.Logs.Type,
-		clusterRequest.includeLegacyForwardConfig(),
-		clusterRequest.includeLegacySyslogConfig(),
-		clusterRequest.useOldRemoteSyslogPlugin(),
-	)
+	op := generator.Options{}
+	if clusterRequest.includeLegacyForwardConfig() {
+		op[generator.IncludeLegacyForwardConfig] = ""
+	}
+	if clusterRequest.includeLegacySyslogConfig() {
+		op[generator.IncludeLegacySyslogConfig] = ""
+	}
+	if clusterRequest.useOldRemoteSyslogPlugin() {
+		op[generator.UseOldRemoteSyslogPlugin] = ""
+	}
 
+	err = fluentd.Verify(&clusterRequest.Cluster.Spec, clusterRequest.OutputSecrets, &clusterRequest.ForwarderSpec, op)
 	if err != nil {
-		log.Error(err, "Unable to create collector config generator")
+		log.Error(err, "Unable to generate log configuration")
 		return "",
 			clusterRequest.UpdateCondition(
 				logging.CollectorDeadEnd,
-				"Unable to generate collector configuration",
-				"No defined logstore destination",
+				"ClusterLogForwarder input validation error",
+				"ClusterLogForwarder input validation error",
 				corev1.ConditionTrue,
 			)
 	}
 
-	clfSpec := &clusterRequest.ForwarderSpec
-	fwSpec := clusterRequest.Cluster.Spec.Forwarder
-
-	generatedConfig, err := generator.Generate(clfSpec, clusterRequest.OutputSecrets, fwSpec)
+	g := generator.MakeGenerator()
+	conf_sections := fluentd.Conf(&clusterRequest.Cluster.Spec, clusterRequest.OutputSecrets, &clusterRequest.ForwarderSpec, op)
```

Comment:
Seems as though this should be hidden in "MakeGenerator()" as at some point we will need to support multiple collectors based on a factory hint

Reply:
this cant be hidden in `MakeGenerator` because the `fluentd` is the package name, not an object which can be returned from a function.
when we support multiple collectors, based on the hint we need to invoke from different package

```
        var collector = clusterRequest.Cluster.Spec.Collection.Logs.Type
	switch collector {
	case logging.LogCollectionTypeFluentd:
		confSections = fluentd.Conf(&clusterRequest.Cluster.Spec, clusterRequest.OutputSecrets, &clusterRequest.ForwarderSpec, op)
	case logging.LogCollectionTypeVector:
		confSections = vector.Conf(&clusterRequest.Cluster.Spec, clusterRequest.OutputSecrets, &clusterRequest.ForwarderSpec, op)
	case "fluent-bit":
	}

```